examples/asset/alter_mesh.rs

```diff
@@ -0,0 +1,231 @@
+//! Shows how to modify mesh assets after spawning.
+
+use bevy::{
+    gltf::GltfLoaderSettings, input::common_conditions::input_just_pressed, prelude::*,
+    render::mesh::VertexAttributeValues, render::render_asset::RenderAssetUsages,
+};
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins)
+        .add_systems(Startup, (setup, spawn_text))
+        .add_systems(
+            Update,
+            alter_handle.run_if(input_just_pressed(KeyCode::Space)),
+        )
+        .add_systems(
+            Update,
+            alter_mesh.run_if(input_just_pressed(KeyCode::Enter)),
+        )
+        .run();
+}
+
+#[derive(Component, Debug)]
+enum Shape {
+    Cube,
+    Sphere,
+}
+
+impl Shape {
+    fn get_model_path(&self) -> String {
+        match self {
+            Shape::Cube => "models/cube/cube.gltf".into(),
+            Shape::Sphere => "models/sphere/sphere.gltf".into(),
+        }
+    }
+
+    fn set_next_variant(&mut self) {
+        *self = match self {
+            Shape::Cube => Shape::Sphere,
+            Shape::Sphere => Shape::Cube,
+        }
+    }
+}
+
+#[derive(Component, Debug)]
+struct Left;
+
+fn setup(
+    mut commands: Commands,
+    asset_server: Res<AssetServer>,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+) {
+    let left_shape = Shape::Cube;
+    let right_shape = Shape::Cube;
+
+    // In normal use, you can call `asset_server.load`, however see below for an explanation of
+    // `RenderAssetUsages`.
+    let left_shape_model = asset_server.load_with_settings(
+        GltfAssetLabel::Primitive {
+            mesh: 0,
+            // This field stores an index to this primitive in its parent mesh. In this case, we
+            // want the first one. You might also have seen the syntax:
+            //
+            //     models/cube/cube.gltf#Scene0
+            //
+            // which accomplishes the same thing.
+            primitive: 0,
+        }
+        .from_asset(left_shape.get_model_path()),
+        // This is the default loader setting, ensuring both `RENDER_WORLD` and `MAIN_WORLD` are
+        // enabled. It's provided explicitly here only by way of demonstration. Unless you have
+        // specific requirements, it's usually safe to use the default.
+        //
+        // When needing to access a `Mesh` asset via `Res<Assets<Mesh>>`, a common mistake is to
+        // use `RENDER_WORLD` by itself, which can cause the asset to be missing from the resource.
+        // (`RENDER_WORLD` without `MAIN_WORLD` will result in the asset being unloaded from the
+        // asset server after it's been sent to the GPU.)
```

Comment:
```suggestion
        // `RenderAssetUsages::all()` is already the default, so this line could be omitted.
        // We leave it in for explicitness however. The `RenderAssetUsages` tell Bevy whether to
        // keep the data around for the GPU, which corresponds to `RenderAssetUsages::RENDER_WORLD`,
        // or for the CPU, which corresponds to `RenderAssetUsages::MAIN_WORLD`. 
        // `RENDER_WORLD` is necessary to render the mesh, `MAIN_WORLD` is necessary to inspect
        // and modify the mesh. Since most games will not need to modify meshes at runtime, many users
        // opt to pass only `RENDER_WORLD`. That is more memory efficient, as we don't need to keep the mesh
        // in the RAM. For this example however, this would not work, as we need to inspect and modify the mesh
        // via `Res<Assets<Mesh>>` at runtime. That's the entire point of the example, after all!
```